### PR TITLE
23629 ess performance improvement spike 2

### PR DESCRIPTION
**Pull request by Siddartha-tech**
Logging improvement with Elapsed

### CODE REVIEW

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Extensions/LoggerExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            Action action,
+            params object[] messageArguments)
+        {
+            logger.LogStartEndAndElapsedTime(startEventId, completedEventId, messageFormat, () =>
+            {
+                action();
+                return 1;
+            }, messageArguments);
+        }
+
```
Comment (barrie-cooper) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Extensions/LoggerExtensions.cs:25]: This is unused and can be removed please
```suggestion
```

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.sln.DotSettings [hunk 2]
```diff
@@ -0,0 +1,3 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
+	<s:Boolean x:Key="/Default/UserDictionary/Words/=catalogue/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/UserDictionary/Words/=UKHO/@EntryIndexedValue">True</s:Boolean></wpf:ResourceDictionary>
```
Comment (barrie-cooper) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.sln.DotSettings:3]: Can we remove this file from the branch please as it's not something we normally have checked in.

**Review by barrie-cooper — COMMENTED** [review]
A couple of minor comments to address please

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Extensions/LoggerExtensions.cs (hunk 1 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Extensions/LoggerExtensions.cs]: Yes. removed method.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.sln.DotSettings (hunk 2 above)
Comment (Siddartha-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.sln.DotSettings]: Yes removed from the branch. Thanks!

**Review by barrie-cooper — APPROVED** [review]
Thanks for changes, LGTM.

**Pull request merged**